servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NonPipelinedCloseHandler.java

```diff
@@ -0,0 +1,183 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.netty.internal;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.transport.netty.internal.CloseHandler.CloseEvent.CHANNEL_CLOSED_INBOUND;
+import static io.servicetalk.transport.netty.internal.CloseHandler.CloseEvent.CHANNEL_CLOSED_OUTBOUND;
+import static io.servicetalk.transport.netty.internal.CloseHandler.CloseEvent.GRACEFUL_USER_CLOSING;
+import static io.servicetalk.transport.netty.internal.CloseHandler.CloseEvent.PROTOCOL_CLOSING_INBOUND;
+import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.State.has;
+import static java.util.Objects.requireNonNull;
+
+final class NonPipelinedCloseHandler extends CloseHandler {
+    private static final byte READ = 0x1;
+    private static final byte WRITE = 0x2;
+    private static final byte IN_CLOSED = 0x4;
+    private static final byte OUT_CLOSED = 0x8;
+    private static final byte CLOSED = 0x10;
+    private static final byte GRACEFUL_CLOSE = 0x20;
+    private static final byte IS_CLIENT = 0x40;
+    private static final byte ALL_CLOSED = IN_CLOSED | OUT_CLOSED | CLOSED;
+    private static final byte READ_WRITE = READ | WRITE;
+    private static final byte CLIENT_IN_WRITE = IS_CLIENT | WRITE | IN_CLOSED;
+    private static final byte GRACEFUL_IN_CLOSED = GRACEFUL_CLOSE | IN_CLOSED;
+    private static final byte GRACEFUL_OUT_CLOSED = GRACEFUL_CLOSE | OUT_CLOSED;
+    private byte state;
+    private Consumer<CloseEvent> eventHandler = __ -> { };
+    @Nullable
+    private CloseEvent closeEvent;
+
+    NonPipelinedCloseHandler(boolean isClient) {
+        if (isClient) {
+            state = IS_CLIENT;
+        }
+    }
+
+    @Override
+    public void protocolPayloadBeginInbound(final ChannelHandlerContext ctx) {
+        state = set(state, READ);
+    }
+
+    @Override
+    public void protocolPayloadEndInbound(final ChannelHandlerContext ctx) {
+        state = unset(state, READ);
+        inboundEventCheckClose(ctx.channel());
+    }
+
+    @Override
+    public void protocolPayloadBeginOutbound(final ChannelHandlerContext ctx) {
+        state = set(state, WRITE);
+    }
+
+    @Override
+    public void protocolPayloadEndOutbound(final ChannelHandlerContext ctx, final ChannelPromise promise) {
+        ctx.pipeline().fireUserEventTriggered(OutboundDataEndEvent.INSTANCE);
+        promise.addListener(f -> {
+            state = unset(state, WRITE);
```

Comment:
`RequestResponseCloseHandler` does this only if `f.isSuccess()` returns `true`. Should we make them consistent and either check the future or ignore its result?

Reply:
this class doesn't need to differentiate between success/failure. I don't understand why we need to check for `success` in `RequestResponseCloseHandler`. The original commit doesn't provide much context https://github.com/apple/servicetalk/commit/7b49eadae705fafadc2304220ec5c392b53ef2ed, and the tests (including the test modified by the commit) pass without this check, I think it should be fine to remove the `success` check and just do it unconditionally as things have changed a lot since the initial change.